modin/experimental/backends/omnisci/query_compiler.py

```diff
@@ -293,20 +294,70 @@ def min(self, **kwargs):
         return self._agg("min", **kwargs)
 
     def sum(self, **kwargs):
+        min_count = kwargs.pop("min_count")
+        if min_count != 0:
+            return super().sum(
+                min_count=min_count,
+                **kwargs,
+            )
         return self._agg("sum", **kwargs)
 
     def mean(self, **kwargs):
         return self._agg("mean", **kwargs)
 
+    def nunique(self, axis=0, dropna=True):
+        if axis != 0 or dropna is False:
+            return super().nunique(axis=axis, dropna=dropna)
+        # TODO: check that the OmniSci server is running with the enabled watchdog
+        if "float" in self.dtypes.values:  # and is_watchdog_enabled():
+            ErrorMessage.single_warning(
+                "'DataFrame.nunique' can't be performed fast enough with the OmniSci backend "
+                "because of the columns with the 'float' type, operation will be defaulted to pandas. "
+                "To use OmniSci implementation get rid of 'float' columns (you may drop those or "
+                "round its values to integer)."
+            )
+            return super().nunique(axis=axis, dropna=dropna)
+
+        return self._agg("count", distinct=True)
+
     def _agg(self, agg, axis=0, level=None, **kwargs):
+        """
+        Perform specified aggregation along rows/columns.
+
+        Parameters
+        ----------
+        agg : str
+            Name of the aggregation function to perform.
+        axis : {0, 1}, default: 0
+            Axis to perform aggregation alogn. 0 is for columns when 1 is for rows.
+            *Note:* OmniSci backend supports aggregation for 0 axis only, aggregation
+            along rows will be defaulted to pandas.
+        level : None
+            Serves the compatibility purpose, always have to be None.
+        **kwargs : dict
+            Additional parameters to pass to the aggregation function.
+
+        Returns
+        -------
+        DFAlgQueryCompiler
+            New single-column (``axis=1``) or single-row (``axis=0``) query compiler containing
```

Comment:
Doesn't this phrase contradict to that one above?
```
Axis to perform aggregation along. 0 is for columns when 1 is for rows.
```

Reply:
No, in case of aggregating along columns the result would be:
```
  col1  col2            col1  col2
0    1     4    ====> 0    6    15   
1    2     5    ====>
2    3     6         
```
so single-row qc is produced.

In the opposite, if `axis=1` then the single-column frame is produced:
```
  col1  col2            0
0    1     4    ====> 0 5   
1    2     5    ====> 1 7
2    3     6          2 9
```

Reply:
It seems to me the docstring (the description of `axis`) looks confused a bit in comparing to native pandas description ([link](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.aggregate.html)). Maybe, we should change it?

Reply:
1. `OmnisciQueryCompiler._agg` is not the equivalent to the high-level `DataFrame.agg` method. It's actually a private OmniSci-backend specific method, which can deviate from the base operation flow.
2. The doc-string about axes show what the function does: 
- if `axis=0` then the function applied to each __column__ and so each column get reduced into a single scalar value, hence frame became a __single-row__
- if `axis=1` then the function applied to each __row__ and so it produces a __single-column__ qc.

Reply:
> 1. `OmnisciQueryCompiler._agg` is not the equivalent to the high-level `DataFrame.agg` method. It's actually a private OmniSci-backend specific method, which can deviate from the base operation flow.
> 2. The doc-string about axes show what the function does:
> 
> * if `axis=0` then the function applied to each **column** and so each column get reduced into a single scalar value, hence frame became a **single-row**
> * if `axis=1` then the function applied to each **row** and so it produces a **single-column** qc.

Okay, I am getting convinced about the docstring's correctness but let's clarify that a bit adding some words about function applied.

Reply:
Expanded axis parameter doc-string at the last commit